timesketch/api/v1/resources/sigma.py

```diff
@@ -0,0 +1,220 @@
+        """Handles GET request to the resource.
+
+        Returns:
+            Dict of sigma rules
+        """
+        sigma_rules = []
+        
+        # TODO: that should be part of the Timesketch config.
+        try:
+            _CONFIG_FILE = current_app.config['SIGMA_CONFIG']
+        except KeyError:
+            logger.error("SIGMA_CONFIG not found in config file")
+            _CONFIG_FILE = '../../../../data/sigma_config.yaml'
+
+        # Path to the directory containing the Sigma Rules to run, relative to
+        # this file.
+        # TODO check if get is better for getting the key errors
+        try:
+            _RULES_PATH = current_app.config['SIGMA_RULES_FOLDER']
+        except KeyError:
+            logger.error("SIGMA_RULES_FOLDER not found in config file")
+            _RULES_PATH = '../../../../data/sigma/rules/'
+
+        sigma_config_path = os.path.join(os.path.dirname(__file__), _CONFIG_FILE)
+
+        with open(sigma_config_path, 'r') as sigma_config_file:
+            sigma_config_file = sigma_config_file.read()
+        sigma_config = sigma_configuration.SigmaConfiguration(sigma_config_file)
+    
+        sigma_backend = sigma_elasticsearch.ElasticsearchQuerystringBackend(
+            sigma_config, {})
+        
+        rules_path = os.path.join(os.path.dirname(__file__), _RULES_PATH)
+
+        for dirpath, dirnames, files in os.walk(rules_path):
+
+            if 'deprecated' in dirnames:
+                dirnames.remove('deprecated')
+
+            for rule_filename in files:
+                if rule_filename.lower().endswith('yml'):
+                    print(rule_filename)
+                    # if a sub dir is found, append it to be scanned for rules
+                    if os.path.isdir(os.path.join(rules_path, rule_filename)):
+                        logger.error(
+                            'this is a directory, skipping: {0:s}'.format(
+                                rule_filename))
+                        continue
+
+                    tag_name, _ = rule_filename.rsplit('.')
+                    rule_file_path = os.path.join(dirpath, rule_filename)
+                    rule_file_path = os.path.abspath(rule_file_path)
+
+                    with codecs.open(rule_file_path, 'r', encoding='utf-8',errors='replace') as rule_file:
+                                try:
+                                    rule_file_content = rule_file.read()
+                                    rule_yaml_data = yaml.safe_load(rule_file_content)
+                                    sigma_rules.append(rule_yaml_data)
+                                except NotImplementedError as exception:
+                                    logger.error(
+                                        'Error generating rule in file {0:s}: {1!s}'
+                                        .format(rule_file_path, exception))
+                                    continue
+        
+        # TODO: is that actually needed?
```

Comment:
yes, we want the object returned ack to have a `meta` and `objects` sections, that's how all other objects are returned and we want there to be a consistency... so please add osmething intot he meta, that is more meaningful than this, which can then be used by the API client to provide some sort of metadata information about the request, adn then have the rest in the `objects` section.

Reply:
ok got it, will change the todo to "add meaningful metadata" and already included the number of rules as one portion to the meta field